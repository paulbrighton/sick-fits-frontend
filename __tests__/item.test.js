import ItemComponent from '../components/Item'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

const fakeItem = {
  id: 'asfghk',
  title: 'Shoes',
  price: 4000,
  description: 'Some shoes',
  image: 'cat.jpg',
  largeImage: 'createHttpLink.jpg'
}

describe('<Item/>', () => {
  it('renders and matches snapshot', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />)
   expect(toJSON(wrapper)).toMatchSnapshot()
  })
  // it('renders price tag and text correctly', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />)
  //   const PriceTag = wrapper.find('PriceTag')
  //   const Title = wrapper.find('Title a')
  //   const Description = wrapper.find('p')
  //   expect(PriceTag.children().text()).toBe('£50')
  //   expect(Title.children().text()).toBe(fakeItem.title)
  //   expect(Description.children().text()).toBe(fakeItem.description)
  // })

  // it('renders the image correctly', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />)
  //   const Image = wrapper.find('img')

  //   expect(Image.props().src).toBe(fakeItem.image)
  //   expect(Image.props().alt).toBe(fakeItem.title)
  // })

  // it('renders the buttons correctly', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />)
  //   const buttonList = wrapper.find('.buttonList')

  //   expect(buttonList.children()).toHaveLength(3)
  //   expect(buttonList.find('Link')).toHaveLength(1)
  //   // Alternative way to find buttons ↑ ↓
  //   expect(buttonList.find('Link').exists()).toBe(true)
  //   expect(buttonList.find('DeleteItem').exists()).toBe(true)
  //   expect(buttonList.find('AddToCart').exists()).toBe(true)
  // })
})
