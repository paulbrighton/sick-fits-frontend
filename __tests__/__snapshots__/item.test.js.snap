// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matches snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="Shoes"
    src="cat.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "asfghk",
          },
        }
      }
    >
      <a>
        Shoes
      </a>
    </Link>
  </Title>
  <PriceTag>
    £40
  </PriceTag>
  <p>
    Some shoes
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "asfghk",
          },
        }
      }
    >
      <a>
        Edit ✏️
      </a>
    </Link>
    <AddToCart
      id="asfghk"
    />
    <DeleteItem
      id="asfghk"
    >
      Delete this Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
